probe/probe.go

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Authors of KubeArmor
+
+package probe
+
+
+import (
+	"fmt"
+	"syscall"
+    "strings"
+	"golang.org/x/mod/semver"
+    "log"
+    "github.com/fatih/color"
+    "os"
+    "github.com/kubearmor/kubearmor-client/k8s"
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func PrintProbeResult(c *k8s.Client) error{
+    
+    checkHostAuditSupport()
+    checkLsmSupport(getHostSupportedLsm())
+    checkNodeAuditSupport(c)
+	return nil
+}
+
+
+func checkLsmSupport(supportedLsm string) {
+    fmt.Printf("\t Enforcement:") 
+    if strings.Contains(supportedLsm, "selinux"){
+        color.Yellow(" Partial")
+    }else if strings.Contains(supportedLsm, "apparmor"){
+        color.Green(" Full")
+    }else{
+        color.Red(" None")
+    }
+}
+
+func getHostSupportedLsm() string {
+
+    b, err := os.ReadFile("/sys/kernel/security/lsm")
+    if err != nil {
+        log.Printf("an error occured when reading file")
+        return "none"
+    }
+    s := string(b)
+    return s
+}
+
+
+
+func kernelVersionSupported(kernelVersion string) bool {
+    if semver.Compare(kernelVersion, "4.14") >= 0 {
+        return true
+    }
+    return false;
+}
+
+
+ // A utility to convert the values to proper strings.
+func int8ToStr(arr []int8) string {
+    b := make([]byte, 0, len(arr))
+    for _, v := range arr {
+        if v == 0x00 {
+            break
+        } 
+        b = append(b, byte(v))
+    }
+    return string(b)
+}
+
+func checkHostAuditSupport() {
+	var uname syscall.Utsname
+    if err := syscall.Uname(&uname); err == nil {
+        kernelVersion:= int8ToStr(uname.Release[:])
+        s := strings.Split(kernelVersion, "-")
+        kversion := s[0]
+        whilte := color.New(color.FgWhite)
+        boldWhite := whilte.Add(color.Bold)
+        boldWhite.Println("Host:")
+        fmt.Printf("\t Observability/Audit:") 
+        if kernelVersionSupported(kversion){
+            color.Green(" Supported (Kernel Version >= 4.14)")
+        }else{
+            color.Red(" Not Supported (Kernel Version < 4.14)")
+        }
+        
+	}else{
+        color.Red(" Error")
+    }
+	
+}
+
+func checkNodeAuditSupport(c *k8s.Client) {
+    whilte := color.New(color.FgWhite)
+    boldWhite := whilte.Add(color.Bold)
```

Comment:
We could declare these outside of function, globally in this package, since being used by multiple function.